src/control/server/mgmt_system_test.go

```diff
@@ -117,6 +119,102 @@ func TestServer_MgmtSvc_LeaderQuery(t *testing.T) {
 	}
 }
 
+func TestServer_MgmtSvc_ClusterEvent(t *testing.T) {
+	rasEventRankFail := ras.NewRankFailEvent(0, 0, nil)
+
+	for name, tc := range map[string]struct {
+		notifyReq *control.SystemNotifyReq
+		sequence  int
+		expResp   *mgmtpb.ClusterEventResp
+		expErr    error
+	}{
+		"nil request": {
+			expErr: errors.New("nil request"),
+		},
+		"nil event": {
+			notifyReq: &control.SystemNotifyReq{},
+			expErr:    errors.New("unexpected event type"),
+		},
+		"successful notification": {
+			notifyReq: &control.SystemNotifyReq{
+				Event: rasEventRankFail,
+			},
+			expErr: errors.New("unexpected event type"),
+		},
```

Comment:
Testing of the processor should be separate from processing the RPC handler. This is going to very quickly get out of hand otherwise.

Reply:
done